db/db_impl/db_impl_files.cc

```diff
@@ -395,8 +403,10 @@ void DBImpl::PurgeObsoleteFiles(JobContext& state, bool schedule_only) {
       state.manifest_delete_files.size());
   // We may ignore the dbname when generating the file names.
   for (auto& file : state.sst_delete_files) {
-    candidate_files.emplace_back(
-        MakeTableFileName(file.metadata->fd.GetNumber()), file.path);
+    if (file.should_delete_file) {
```

Comment:
I find the name here confusing: if a file is already in `state_delete_files`, why will `should_delete_file` be false? Only when I checked the definition of `should_delete_file` did I find out. I wonder whether we can have a better name than `should_delete_file`.